src/main/groovy/liquibase/sdk/test/change/ChangeObjectTests.groovy

```diff
@@ -0,0 +1,153 @@
+package liquibase.sdk.test.change
+
+import groovy.transform.builder.Builder
+import liquibase.CatalogAndSchema
+import liquibase.Liquibase
+import liquibase.database.Database
+import liquibase.database.jvm.JdbcConnection
+import liquibase.sdk.test.config.DatabaseUnderTest
+import liquibase.sdk.test.config.TestConfig
+import liquibase.sdk.test.util.FileUtils
+import liquibase.sdk.test.util.SnapshotHelpers
+import liquibase.sdk.test.util.TestUtils
+import liquibase.util.StringUtil
+import org.skyscreamer.jsonassert.JSONAssert
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import java.util.logging.Logger
+
+class ChangeObjectTests extends Specification {
+
+    final static List supportedChangeLogFormats = ['xml', 'sql', 'json', 'yml', 'yaml'].asImmutable()
+
+    @Unroll
+    def "apply #testInput.changeObject against #testInput.databaseName; verify generated SQL and DB snapshot"() {
+        given:
+        Liquibase liquibase = TestUtils.createLiquibase(testInput.pathToChangeLogFile, testInput.database)
+
+        String expectedSql = cleanSql(FileUtils.getFileContent(testInput.changeObject, testInput.database.shortName, testInput.database.databaseMajorVersion, testInput.database.databaseMinorVersion, "liquibase/sdk/test/change/expectedSql", ".sql"))
```

Comment:
database minor and major version provided by jdbc driver might be incorrect. This happens with cockroachdb. It uses postgre jdbc driver and versions provided by driver is incorrect. For that reason we use version from yaml as main source of truth, and the on provided by jdbc driver only if version in yaml is not provided